### PR TITLE
Restore test mode when SPARSEZOO_TEST_MODE is true

### PR DESCRIPTION
**Pull request by dbogunowicz**
Testing:
Together with @kazukageorge , we tested in staging/prod that:
- there are entries in the db when I run the test suite with `SPARSEZOO_TEST_MODE=false`
- there are no entries in the db when I run the test suite with `SPARSEZOO_TEST_MODE=true`

### CODE REVIEW

File: src/sparsezoo/utils/requests.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 import requests
 
-from . import MODELS_API_URL
+from . import MODELS_API_URL, SPARSEZOO_TEST_MODE
```
Comment (bfineran) [inline src/sparsezoo/utils/requests.py:22]: nit: remove relative import (line below as well)

File: src/sparsezoo/utils/requests.py [hunk 2]
```diff
@@ -106,15 +106,20 @@ def download_get_request(
...
         download_args.append(f"release_version={args.release_version}")
 
+    if SPARSEZOO_TEST_MODE:
+        # important, do not remove
+        download_args.append("increment_download=False")
+
     if download_args:
         url = f"{url}?{'&'.join(download_args)}"
 
+    print(url)
```
Comment (bfineran) [inline src/sparsezoo/utils/requests.py:116]: debugging?

**Comment by dbogunowicz** [comment]
There is still some uncertainty about whether tests do not "leak" when ran in GHA. I propose we land this PR and me with @kazukageorge look into GHA later.

**Comment by kazukageorge** [comment]
So it passed on Damian's local testing machine (did not write to staging and prod db with tests), but it looks like github actions causing the write to the db. Hence, code should be fine, might be some set up (ex. env vars) issue on github actions' machine

File: src/sparsezoo/utils/requests.py [hunk 3]
```diff
@@ -106,6 +106,10 @@ def download_get_request(
     if hasattr(args, "release_version") and args.release_version:
         download_args.append(f"release_version={args.release_version}")
 
+    if SPARSEZOO_TEST_MODE:
+        # important, do not remove
+        download_args.append("increment_download=False")
```
Comment (corey-nm) [inline src/sparsezoo/utils/requests.py:111]: Another way we could do this in the future is by setting the User-Agent header to something specific to unit tests in this case, and then the server could check against that specific User-Agent. Then we wouldn't need a custom flag!

**Review by corey-nm — APPROVED** [review]

File: src/sparsezoo/utils/helpers.py [hunk 4]
```diff
@@ -14,9 +14,28 @@
...
+__all__ = [
+    "create_dirs",
+    "create_parent_dirs",
+    "clean_path",
+    "remove_tar_duplicates",
+    "convert_to_bool",
+]
+
+
+def convert_to_bool(val: Any):
```
Comment (corey-nm) [inline src/sparsezoo/utils/helpers.py:29]: Maybe add some unit tests for this one? Seems like it should be pretty straightforward to test

File: src/sparsezoo/utils/__init__.py [hunk 5]
```diff
@@ -17,12 +17,14 @@
...
 import os
 
+from sparsezoo.utils.helpers import convert_to_bool
 
 BASE_API_URL = (
     os.getenv("SPARSEZOO_API_URL")
     if os.getenv("SPARSEZOO_API_URL")
     else "https://api.neuralmagic.com"
 )
+SPARSEZOO_TEST_MODE = convert_to_bool(os.getenv("SPARSEZOO_TEST_MODE"))
```
Comment (corey-nm) [inline src/sparsezoo/utils/__init__.py:27]: Actually can we move this computation down into the actual function? That way the env variable is checked every time the function is called so we aren't dealing with some weird global variable caching

**Comment by corey-nm** [comment]
> So it passed on Damian's local testing machine (did not write to staging and prod db with tests), but it looks like github actions causing the write to the db. Hence, code should be fine, might be some set up (ex. env vars) issue on github actions' machine

Which repo's github actions are you looking at? It seems like sparsezoo and sparseml both set the environment variable properly, but deepsparse does not. Could the updates be coming from deepsparse workflows?

**Comment by kazukageorge** [comment]
> > So it passed on Damian's local testing machine (did not write to staging and prod db with tests), but it looks like github actions causing the write to the db. Hence, code should be fine, might be some set up (ex. env vars) issue on github actions' machine
> 
> Which repo's github actions are you looking at? It seems like sparsezoo and sparseml both set the environment variable properly, but deepsparse does not. Could the updates be coming from deepsparse workflows?

Thanks for the reply.

I re-ran the tests and it passed (did not write to the db). Something else was running independently causing the write to the db.

File: src/sparsezoo/utils/requests.py (hunk 3 above)
Comment (bfineran) [reply src/sparsezoo/utils/requests.py:112]: good idea, especially for outside testers /devs, but for now, still need ways to disable download increments for internal use + manual QA runs

**Review by corey-nm — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Review by KSGulin — APPROVED** [review]

**Pull request merged**